sklearn/cluster/_bisect_k_means.py

```diff
@@ -0,0 +1,406 @@
+import warnings
+
+import numpy as np
+
+from sklearn.utils.validation import check_array, check_is_fitted, \
+    _check_sample_weight, check_random_state
+
+from ..utils.extmath import row_norms
+
+from ._kmeans import KMeans, _kmeans_single_elkan, \
+    _kmeans_single_lloyd, _labels_inertia_threadpool_limit
+
+import scipy.sparse as sp
+from sklearn.exceptions import ConvergenceWarning, EfficiencyWarning
+from ._kmeans import _kmeans_plusplus
+
+
+class BisectKMeans(KMeans):
+    """ Bisecting K-Means clustering
```

Comment:
Please tell us here in one short sentence how it differs from standard K-Means, so that the user understand what this model does even if he or she does not know the name.

Reply:
Also, the docstring (and the docs) should reference in several places that this is a hierarchical clustering tool (some people with be looking for clustering tools with this keyword).

Reply:
Maybe not in one line, but done. @GaelVaroquaux would you might to have a look now?

Reply:
Thanks, that's super useful!!